src/main/java/LinkedBinarySearchTree.java

```diff
@@ -0,0 +1,492 @@
+            sequence.add(current.getData());
+        }
+    }
+
+    /**
+     * Level order traversal of a binary tree. Elements are added
+     * to a bag in the required order.
+     *
+     * @param start    Root node of a binary tree
+     * @param sequence Bag containing elements in level order order
+     */
+    private void levelOrder(Node<T> start, IndexedBag<T> sequence) {
+        Queue<Node<T>> levelOrderQueue = new ArrayQueue<>();
+        Node<T> current;
+        levelOrderQueue.enqueue(start);
+        while (!levelOrderQueue.isEmpty()) {
+            current = levelOrderQueue.dequeue();
+            sequence.add(current.getData());
+            if (current.getLeft() != null) {
+                levelOrderQueue.enqueue(current.getLeft());
+            }
+            if (current.getRight() != null) {
+                levelOrderQueue.enqueue(current.getRight());
+            }
+        }
+    }
+
+    @Override
+    public Iterator<T> iterator() {
+        return inOrderIterator();
+    }
+
+    @Override
+    public Iterator<T> preOrderIterator() {
+        IndexedBag<T> sequence = new ArrayIndexedBag<>();
+        preOrder(root, sequence);
+        return sequence.iterator();
+    }
+
+    @Override
+    public Iterator<T> inOrderIterator() {
+        IndexedBag<T> sequence = new ArrayIndexedBag<>();
+        inOrder(root, sequence);
+        return sequence.iterator();
+    }
+
+    @Override
+    public Iterator<T> postOrderIterator() {
+        IndexedBag<T> sequence = new ArrayIndexedBag<>();
+        postOrder(root, sequence);
+        return sequence.iterator();
+    }
+
+    @Override
+    public Iterator<T> levelOrderIterator() {
+        IndexedBag<T> sequence = new ArrayIndexedBag<>();
+        if (!isEmpty()) {
+            levelOrder(root, sequence);
+        }
+        return sequence.iterator();
+    }
+
+    /**
+     * Returns a string containing the elements in the binary search tree in
+     * an in order order.
+     *
+     * @return String of the elements in in order order
+     */
+    @Override
+    public String toString() {
+        StringBuilder builder = new StringBuilder();
+        for (T element : this) {
+            builder.append(element);
+            builder.append(", ");
+        }
+        return builder.toString();
+    }
+
+    /**
+     * A Node class for a binary linked tree structure. Each node
+     * contains a reference to data of type T, which may be
+     * null, and a reference to the left and right child
+     * binary linked nodes, which may also be null.
+     * <p>
+     * This class is a static nested class since the node
+     * class is only needed for the implementation of the
+     * LinkedBinarySearchTree.
+     *
+     * @param <T> Type of the data being stored in the node
+     */
+    private static class Node<T> {
+
+        private T data;
+        private Node<T> left;
+        private Node<T> right;
+
+        public Node() {
+            this(null);
+        }
```

Comment:
private class, public methods (with no interfaces being implemented). That's called overexposure. `public` -> `private` on all these methods.

Related, the default constructor is not used (which your IDE would yell about if the methods were not overexposed).